org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/impl/XbaseInterpreter.java

```diff
@@ -294,8 +294,13 @@ protected Object _doEvaluate(XNullLiteral literal, IEvaluationContext context, C
 	}
 	
 	protected Object _doEvaluate(XReturnExpression returnExpr, IEvaluationContext context, CancelIndicator indicator) {
-		Object returnValue = internalEvaluate(returnExpr.getExpression(), context, indicator);
-		throw new ReturnValue(returnValue);
+		XExpression expression = returnExpr.getExpression();
+		if (expression != null) {
+			Object returnValue = internalEvaluate(returnExpr.getExpression(), context, indicator);
```

Comment:
wonder if this is a regression in internalEvaluate

Reply:
and or if we should have an elseif for Null/Void in  org.eclipse.xtext.xbase.interpreter.impl.XbaseInterpreter.doEvaluate(XExpression, IEvaluationContext, CancelIndicator)

Reply:
@szarnekow what do you think?

Reply:
> wonder if we should add a testcase to
> AbstractXbaseEvaluationTest / XbaseInterpreterTest

Yes, that would be necessary. Xbase does not allow to define methods though the tests should be based on lambda expressions with a procedural type, e.g. `val Runnable r = [ return; ]; r.run()`

Reply:
I don't think it's a regression. Looks like it was an incomplete solution since 2011

Reply:
> an elseif for Null/Void in org.eclipse.xtext.xbase.interpreter.impl.XbaseInterpreter.doEvaluate

No, I don't think so. A null expression would be a programming error which is already indicated by the IllegalArgumentException.